resource/updater.go

```diff
@@ -0,0 +1,54 @@
+package resource
+
+import (
+	"context"
+	"errors"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// UpdateObject ensures an object is updated by calling Get on the provided client to get the current state of the object,
+// then runs updateFunc on it to update it, and finally calls UpdateInto with the updated object and provided config,
+// using the ResourceVersion of the updated object. If the update fails due to a 409/Conflict, it will retry the whole process up to two times.
+func UpdateObject[T Object](ctx context.Context, client Client, identifier Identifier, updateFunc func(obj *T, isRetry bool) error, opts UpdateOptions) (T, error) {
```

Comment:
The pointer-to-pointer semantics make this very confusing IMO. Just for that I would change the signature of the function to `(T, bool) -> (T, error)`.